File: test_ai_helpers.rs
#!/usr/bin/env rust-script

//! Simple test runner for AI helpers functionality
//! This avoids the full test suite compilation issues

use std::process::Command;
use std::time::Duration;

fn main() {
    println!("🧪 Testing AI Helper Ecosystem...");
    
    // Test basic compilation
    println!("📦 Testing compilation...");
    let compile_result = Command::new("cargo")
        .args(&["check", "--lib"])
        .output()
        .expect("Failed to run cargo check");
    
    if !compile_result.status.success() {
        println!("❌ Compilation failed!");
        println!("STDERR: {}", String::from_utf8_lossy(&compile_result.stderr));
        return;
    }
    
    println!("✅ Compilation successful!");
    
    // Test Python dependencies
    println!("🐍 Testing Python dependencies...");
    let python_result = Command::new("python3")
        .args(&["-c", "import torch, transformers, sentence_transformers, chromadb; print('✅ Python dependencies OK')"])
        .output()
        .expect("Failed to test Python dependencies");
    
    if !python_result.status.success() {
        println!("⚠️  Python dependencies not fully available");
        println!("STDERR: {}", String::from_utf8_lossy(&python_result.stderr));
        println!("💡 Try: pip3 install -r python/requirements.txt");
    } else {
        println!("{}", String::from_utf8_lossy(&python_result.stdout));
    }
    
    // Test Python model service script
    println!("🤖 Testing Python model service...");
    let model_service_result = Command::new("python3")
        .args(&["-c", r#"
import sys
sys.path.append('python')
try:
    from model_service import ModelService
    service = ModelService()
    print('✅ Python model service can be imported')
except Exception as e:
    print(f'❌ Python model service error: {e}')
"#])
        .output()
        .expect("Failed to test Python model service");
    
    println!("{}", String::from_utf8_lossy(&model_service_result.stdout));
    
    // Test basic struct creation
    println!("🏗️  Testing Rust struct creation...");
    let struct_test = Command::new("cargo")
        .args(&["test", "--lib", "test_ai_helper_creation", "--", "--nocapture"])
        .output()
        .expect("Failed to run struct test");
    
    if struct_test.status.success() {
        println!("✅ Rust struct creation test passed!");
    } else {
        println!("⚠️  Rust struct test had issues:");
        println!("STDERR: {}", String::from_utf8_lossy(&struct_test.stderr));
    }
    
    println!("\n🎉 AI Helper Ecosystem basic tests completed!");
    println!("💡 For full integration tests, ensure Python dependencies are installed and run:");
    println!("   cargo test ai_helpers_e2e_test --release");
}

File: test_ai_helpers.py
#!/usr/bin/env python3

"""
Simple test runner for AI helpers functionality
This avoids the full test suite compilation issues
"""

import subprocess
import sys
import os

def run_command(cmd, description):
    """Run a command and return success status"""
    print(f"🔧 {description}...")
    try:
        result = subprocess.run(cmd, shell=True, capture_output=True, text=True, timeout=60)
        if result.returncode == 0:
            print(f"✅ {description} successful!")
            if result.stdout.strip():
                print(f"Output: {result.stdout.strip()}")
            return True
        else:
            print(f"❌ {description} failed!")
            if result.stderr.strip():
                print(f"Error: {result.stderr.strip()}")
            return False
    except subprocess.TimeoutExpired:
        print(f"⏰ {description} timed out!")
        return False
    except Exception as e:
        print(f"💥 {description} error: {e}")
        return False

def main():
    print("🧪 Testing AI Helper Ecosystem...")
    
    # Test basic compilation
    success = run_command("cargo check --lib", "Testing compilation")
    if not success:
        return 1
    
    # Test Python dependencies
    print("🐍 Testing Python dependencies...")
    
    # Test individual dependencies
    deps = [
        ("torch", "PyTorch"),
        ("transformers", "Transformers"),
        ("sentence_transformers", "Sentence Transformers"),
        ("chromadb", "ChromaDB"),
        ("numpy", "NumPy"),
    ]
    
    missing_deps = []
    for dep, name in deps:
        cmd = f"python3 -c 'import {dep}; print(f\"✅ {name} available\")"
        if not run_command(cmd, f"Testing {name}"):
            missing_deps.append(dep)
    
    if missing_deps:
        print(f"⚠️  Missing dependencies: {', '.join(missing_deps)}")
        print("💡 Try: pip3 install -r python/requirements.txt")
    
    # Test Python model service script
    print("🤖 Testing Python model service...")
    
    if os.path.exists("python/model_service.py"):
        test_code = """
import sys
sys.path.append('python')
try:
    from model_service import ModelService
    service = ModelService()
    print('✅ Python model service can be imported')
except Exception as e:
    print(f'❌ Python model service error: {e}')
"""
        with open("test_model_service.py", "w") as f:
            f.write(test_code)
        
        run_command("python3 test_model_service.py", "Testing Python model service")
        
        # Cleanup
        if os.path.exists("test_model_service.py"):
            os.remove("test_model_service.py")
    else:
        print("⚠️  Python model service not found")
    
    # Test basic struct creation (if compilation worked)
    print("🏗️  Testing Rust AI helper creation...")
    run_command("cargo test --lib test_ai_helper_creation -- --nocapture", "Testing Rust struct creation")
    
    # Test configuration files
    print("📋 Testing configuration files...")
    
    config_files = [
        "python/requirements.txt",
        "python/model_service.py",
        "src/ai_helpers/mod.rs",
        "src/ai_helpers/python_models.rs",
    ]
    
    for config_file in config_files:
        if os.path.exists(config_file):
            print(f"✅ {config_file} exists")
        else:
            print(f"❌ {config_file} missing")
    
    print("\n🎉 AI Helper Ecosystem basic tests completed!")
    print("💡 For full integration tests, ensure Python dependencies are installed and run:")
    print("   cargo test ai_helpers_e2e_test --release")
    
    return 0

if __name__ == "__main__":
    sys.exit(main())